src/ec/eddsa.rs

```diff
@@ -108,36 +86,28 @@ impl<'a> Ed25519KeyPair {
     /// corruption that might have occurred during storage of the key pair.
     pub fn from_bytes(private_key: &[u8], public_key: &[u8])
                       -> Result<Ed25519KeyPair, error::Unspecified> {
-        let pair = try!(Ed25519KeyPair::from_bytes_unchecked(private_key,
-                                                             public_key));
+        let private_key = try!(slice_as_array_ref!(private_key, SEED_LEN));
+        let public_key =
+            try!(slice_as_array_ref!(public_key, PUBLIC_KEY_LEN));
+
+        let pair = Ed25519KeyPair::from_bytes_unchecked(private_key, public_key);
         let mut public_key_check = [0; PUBLIC_KEY_LEN];
-        public_from_private(&pair.private_key, &mut public_key_check);
-        if public_key != public_key_check {
+        public_from_private(&private_key, &mut public_key_check);
+        if &public_key[..] != &public_key_check[..] {
```

Comment:
`[u8; 32]` doesn't have operator `!=` implemented for it. Basically this is just a workaround for the lack of generics parameterized by array length.